osg/short-term.md

```diff
@@ -0,0 +1,292 @@
+will be mounted to this path.
+
+## Ticket List Files
+
+The input and output ticket list files are comma-delimited files with each significant line containing an iRODS ticket
+string and a file path. Blank lines and lines beginning with a hash mark (`#`) are ignored. The first line in the file
+is a comment line containing the MIME type, `application/vnd.de.tickets-path-list+csv; version=1`.
+
+As mentioned above, each significatn line in the file contains the iRODS ticket string followed by a comma and the path
+corresponding to the ticket. The ticket string is guaranteed not to contain commas, but the path may. None of the fields
+in the ticket list file are ever quoted. Instead, parsers must rely on the fact that tickets will never contain commas
+to parse the field. For example, suppose the following line is found in a ticket list file:
+
+```
+521CDB78-8EA4-4F14-94FF-D506DB0D45D7,/iplant/home/nobody/the foo, the bar, and the baz
+```
+
+In this example the ticket string is `521CDB78-8EA4-4F14-94FF-D506DB0D45D7` and the path is `/iplant/home/nobody/the
+foo, the bar, and the baz`.
+
+The [wrapper script reference implementation](2) contains a convenient context manager class that can be used to
+correctly parse a ticket list file:
+
+``` python
+# This is a simple context manager class designed to make it easier to read and write iRODS ticket list files.
+class TicketListReader:
+    def __init__(self, path):
+        self.path = path
+
+    def __enter__(self):
+        self.fd = open(self.path, "r")
+        self.r = itertools.imap(lambda l: l.strip().split(",", 1), itertools.ifilter(lambda l: l[0] != '#', self.fd))
+        return self.r
+
+    def __exit__(self, type, value, traceback):
+        self.fd.close()
+```
+
+## Job Config Files
+
+The job configuration file is a JSON file named `config.json` containing configuration settings for the job. The
+following configuration settings are provided:
+
+### arguments
+
+This setting is an array of strings containing the command-line arguments generated by the DE based on the app
+configuration and the parameter values selected by the user. In cases where the wrapper script runs the tool itself as a
+subprocess, this argument list can generally be passed to the subprocess as-is. In other cases, the wrapper script must
+interpret the argument array directly.
+
+### irods_host
+
+This setting contains the host name to use when connecting to iRODS.
+
+### irods_port
+
+This setting contains the port number to use when connecting to iRODS.
+
+### irods_job_user
+
+This setting contains the username of the person who submitted the job. This user should have ownership permissions on
+the output files after the job completes.
+
+### irods_user
+
+This setting contains the username to use when authenticating to iRODS. This setting can be ignored if you're using
+custom iRODS credentials.
```

Comment:
We might want to mention the option for using custom credentials above in the list of tasks the app integrator performs; currently that list only mentions using the credentials provided in the config.json and we wouldn't want implementors to have to read the whole way down to here to figure out they can use their own credentials if they want!

Reply:
Good point. I'll mention it above as well.